src/wasm.h

```diff
@@ -273,6 +286,382 @@ class Literal {
     restoreNormalColor(o);
     return o << ')';
   }
+
+  Literal countLeadingZeroes() const {
+    if (type == WasmType::i32) return Literal((int32_t)CountLeadingZeroes(i32));
+    if (type == WasmType::i64) return Literal((int64_t)CountLeadingZeroes(i64));
+    WASM_UNREACHABLE();
+  }
+  Literal countTrailingZeroes() const {
+    if (type == WasmType::i32) return Literal((int32_t)CountTrailingZeroes(i32));
+    if (type == WasmType::i64) return Literal((int64_t)CountTrailingZeroes(i64));
+    WASM_UNREACHABLE();
+  }
+  Literal popCount() const {
+    if (type == WasmType::i32) return Literal((int32_t)PopCount(i32));
+    if (type == WasmType::i64) return Literal((int64_t)PopCount(i64));
+    WASM_UNREACHABLE();
+  }
+
+  Literal extendToSI64() const {
+    assert(type == WasmType::i32);
+    return Literal((int64_t)i32);
+  }
+  Literal extendToUI64() const {
+    assert(type == WasmType::i32);
+    return Literal((uint64_t)(uint32_t)i32);
+  }
+  Literal extendToF64() const {
+    assert(type == WasmType::f32);
+    return Literal(double(getf32()));
+  }
+  Literal truncateToI32() const {
+    assert(type == WasmType::i64);
+    return Literal((int32_t)i64);
+  }
+  Literal truncateToF32() const {
+    assert(type == WasmType::f64);
+    return Literal(float(getf64()));
```

Comment:
There are spec tests for this, yes. But I don't know offhand what they do, just that this passes them ;)